server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1402,6 +1402,18 @@ private void closeConsistencyCheck() {
         throw new RuntimeException(msg);
       }
 
+      if (tabletMeta.getFlushId().orElse(-1) != lastFlushID) {
+        String msg = "Closed tablet " + extent + " has not been flushed before being closed.";
```

Comment:
The flush status is uncertain here, all we know is that this in memory data is inconsistent with the metadata table.

```suggestion
        String msg = "Closed tablet " + extent + " lastFlushId is inconsistent with metadata : "+(tabletMeta.getFlushId().orElse(-1)+" !=" +lastFlushID;
```